### PR TITLE
feat(suite): Share indicator

### PR DESCRIPTION
**Pull request by jvaclavik**
## Related Issue

Resolve 

## Screenshots:
image

https://github.com/trezor/trezor-suite/assets/858321/766b153f-5017-4d69-975a-4680f09cc515

### CODE REVIEW

File: packages/suite/src/views/dashboard/components/AssetsView/components/AssetInfo.tsx [hunk 1]
```diff
@@ -62,17 +68,29 @@ const LogoWrapper = styled.div`
...
+          )?.fiatPercentage
+        : undefined;
+
     const theme = useTheme();
 
     return (
         <Container onClick={onClick}>
             <LogoWrapper>
-                <CoinLogo symbol={symbol} size={24} hasShareIndicator />
+                <CoinLogo
```
Comment (dahaca) [inline packages/suite/src/views/dashboard/components/AssetsView/components/AssetInfo.tsx:89]: 1. Maybe split `CoinLogo` into 2 components? Just the logo and the `AssetShare` or smth?

File: packages/suite/src/views/dashboard/components/AssetsView/components/AssetInfo.tsx [hunk 2]
```diff
@@ -62,17 +68,29 @@ const LogoWrapper = styled.div`
...
 `;
 
-export const AssetInfo = ({ network, onClick }: AssetInfoProps) => {
+export const AssetInfo = ({ network, onClick, assetsFiatBalance }: AssetInfoProps) => {
     const { symbol, name } = network;
     const selectedAccounts = useSelector((state: any) =>
         selectAccountsByNetworkSymbol(state, symbol),
     );
+
+    const assetPercentage = assetsFiatBalance
```
Comment (dahaca) [inline packages/suite/src/views/dashboard/components/AssetsView/components/AssetInfo.tsx]: I would ideally move this into the theoretical `AssetShare` component, although having `@suite-common/assets` as a dependancy might not be super pretty... Wdyt?

File: packages/suite/src/views/dashboard/components/AssetsView/index.tsx [hunk 3]
```diff
@@ -125,7 +132,21 @@ export const AssetsView = () => {
             const assetFailed = accounts.find(f => f.symbol === network.symbol && f.failed);
             return { symbol, network, assetFailed: !!assetFailed, assetBalance };
         })
-        .filter(data => data !== null) as assetType[];
+        .filter(data => data !== null) as AssetType[];
+
+    const assetsFiatBalance = assetsData.reduce<AssetFiatBalance[]>((acc, asset) => {
```
Comment (dahaca) [inline packages/suite/src/views/dashboard/components/AssetsView/index.tsx]: nit: `assetsFiatBalances`?

File: packages/suite/src/views/dashboard/components/AssetsView/index.tsx [hunk 4]
```diff
@@ -18,6 +18,17 @@ import { useEnabledNetworks } from 'src/hooks/settings/useEnabledNetworks';
 import { AssetCard, AssetCardSkeleton } from './components/AssetCard';
 import { AssetRow, AssetRowSkeleton } from './components/AssetRow';
 import { spacingsPx, typography } from '@trezor/theme';
+import { AssetFiatBalance } from '@suite-common/assets';
+import { toFiatCurrency } from '@suite-common/wallet-utils';
+import { selectLocalCurrency } from 'src/reducers/wallet/settingsReducer';
+import { Network } from '@suite-common/wallet-config';
+
+interface AssetType {
```
Comment (dahaca) [inline packages/suite/src/views/dashboard/components/AssetsView/index.tsx:26]: very similar to the native one? 🤔

File: packages/suite/src/views/dashboard/components/AssetsView/index.tsx [hunk 5]
```diff
@@ -74,20 +85,16 @@ const StyledIcon = styled(Icon)`
...
 
-interface assetType {
-    symbol: string;
-    network: Network;
-    assetBalance: BigNumber;
-    assetFailed: boolean;
-}
-
 export const AssetsView = () => {
     const theme = useTheme();
```
Comment (dahaca) [inline packages/suite/src/views/dashboard/components/AssetsView/index.tsx:112]: nit: put all the selectors at the top, to make the component structure more predictable.  Code style [article](https://www.notion.so/satoshilabs/Components-9159b639d21b43ef8b66be150854b6b6?pvs=4#308aa7db519e4f4eaf70312f60ce030a)

File: packages/suite/src/views/dashboard/components/AssetsView/index.tsx (hunk 3 above)
Comment (dahaca) [inline packages/suite/src/views/dashboard/components/AssetsView/index.tsx]: maybe memo this?

Comment (dahaca) [reply packages/suite/src/views/dashboard/components/AssetsView/index.tsx]: also meybe defice that function outside of the component for better readability?

**Review by dahaca — COMMENTED** [review]
I have [changed](https://github.com/trezor/trezor-suite/pull/10261/commits/af1639866b4a816e834c7ad044da8441ba417cf1) the way the progress circle is made because I was not a fan of `requestAnimationFrame` and changing a prop super fast… Rewrote with svgs + `framer-motion` – it’s smoother, with a better animation :pray:

File: packages/suite/src/views/dashboard/components/AssetsView/index.tsx (hunk 4 above)
Comment (jvaclavik) [reply packages/suite/src/views/dashboard/components/AssetsView/index.tsx:26]: Yes it is. I tried to unify it but it's not easy. For example on mobile they use different type for `network`. Also I think `calculateAssetsPercentage` should not expect complex type with props that are not used.

File: packages/suite/src/views/dashboard/components/AssetsView/components/AssetInfo.tsx (hunk 1 above)
Comment (jvaclavik) [reply packages/suite/src/views/dashboard/components/AssetsView/components/AssetInfo.tsx:89]: Yes, I'm working on this in Account menu issue

**Comment by jvaclavik** [comment]
> I have [changed](https://github.com/trezor/trezor-suite/pull/10261/commits/af1639866b4a816e834c7ad044da8441ba417cf1) the way the progress circle is made because I was not a fan of `requestAnimationFrame` and changing a prop super fast… Rewrote with svgs + `framer-motion` – it’s smoother, with a better animation 🙏

Are you sure it's working? 

I tried to fake it:
image

Values on input are correct:
image

But share indicators not:
image

I'm pretty sure it worked before.


Btw why don't you like `requestAnimationFrame`? I feel like it's easy to understand since it's a standardized way how to animate without 3rd party library.

**Comment by jvaclavik** [comment]
> onst assetsFiatBalance = assetsData.reduce<AssetFiatBalance[]>((acc, asset) => {

I get why it's happening, I will fix it:

image

File: packages/suite/src/views/dashboard/components/AssetsView/components/AssetInfo.tsx (hunk 2 above)
Comment (jvaclavik) [reply packages/suite/src/views/dashboard/components/AssetsView/components/AssetInfo.tsx]: I don't think it's necessary for now. Let's fix it later when CoinLogo refactoring will happen.

File: packages/suite/src/views/dashboard/components/AssetsView/index.tsx (hunk 4 above)
Comment (dahaca) [reply packages/suite/src/views/dashboard/components/AssetsView/index.tsx:26]: Thought we can connect the mobile an desktop UI throught types at least. but it's just a though

**Comment by dahaca** [comment]
> Btw why don't you like `requestAnimationFrame`? I feel like it's easy to understand since it's a standardized way how to animate without 3rd party library.

I've been trying to move animations which cannot be done with css to `framer-motion`, which is cleaner IMO than using `requestAnimationFrame` manually, adding `refs` etc. In this particular example my approach is more complex. That is because I wanted a smooth animation, which theoretically can be done with applying a velocity modifier over time in your approach to avoid 3rd parties but that would be too much, don't you think? 😅 But also because I needed an SVG, since what i have achieved cannot be done via CSS. And finally, `framer-motion` does not update the React component itself, and updating component state so many times a second seemed off to me 😓 Probably no visible performance hit but also I havent tested it with 6 cons on a weaker machine.

**Review by dahaca — APPROVED** [review]

**Pull request merged**